cmake/hex.cmake

```diff
@@ -1,56 +0,0 @@
-# SPDX-License-Identifier: Apache-2.0
-
-# from https://gist.github.com/korzo89/71a6de0f388f7cf8b349101b0134060c
-function(from_hex HEX DEC)
-    string(SUBSTRING "${HEX}" 2 -1 HEX)
-    string(TOUPPER "${HEX}" HEX)
-    set(_res 0)
-    string(LENGTH "${HEX}" _strlen)
-
-    while(_strlen GREATER 0)
-        math(EXPR _res "${_res} * 16")
-        string(SUBSTRING "${HEX}" 0 1 NIBBLE)
-        string(SUBSTRING "${HEX}" 1 -1 HEX)
-        if(NIBBLE STREQUAL "A")
-            math(EXPR _res "${_res} + 10")
-        elseif(NIBBLE STREQUAL "B")
-            math(EXPR _res "${_res} + 11")
-        elseif(NIBBLE STREQUAL "C")
-            math(EXPR _res "${_res} + 12")
-        elseif(NIBBLE STREQUAL "D")
-            math(EXPR _res "${_res} + 13")
-        elseif(NIBBLE STREQUAL "E")
-            math(EXPR _res "${_res} + 14")
-        elseif(NIBBLE STREQUAL "F")
-            math(EXPR _res "${_res} + 15")
-        else()
-            math(EXPR _res "${_res} + ${NIBBLE}")
-        endif()
-
-        string(LENGTH "${HEX}" _strlen)
-    endwhile()
-
-    set(${DEC} ${_res} PARENT_SCOPE)
-endfunction()
-
-function(to_hex DEC HEX)
```

Comment:
no problem in changing the implementation of this, but we should deprecate the function before removing it.

There might be out-of-tree users relying on this function.

Please keep the function, and make an update in `FindDeprecated()` so we remember to remove the function after two releases:
https://github.com/zephyrproject-rtos/zephyr/blob/9074feabf6f827f6b37a69d77a5e13093799a905/cmake/modules/FindDeprecated.cmake#L5-L16

Reply:
Thanks for the comment - I'll get on this, hopefully later today.

Reply:
I'm not 100% clear on how to action this... Obviously we'd like a warning printed when anyone uses the `to_hex()` and `from_hex()` functions, but moving them to `cmake/modules/FindDeprecated.cmake` will make them "_disappear_" for anyone currently using them.

Additionally, we can't easily use the `find_package(Deprecated COMPONENTS ...)` used elsewhere, because `cmake/modules/version.cmake` seems to be included very early on with `CMAKE_MODULE_PATH` unset (and thus `.../FindDeprecated.cmake` is not found).

I'm going to propose a `message(DEPRECATION ...)` in `cmake/hex.cmake`, which is presented when it's included. I hope that suffices.

Reply:
> because cmake/modules/version.cmake seems to be included very early on with CMAKE_MODULE_PATH unset (and thus .../FindDeprecated.cmake is not found).

but it's not the `version.cmake` which should call `FindDeprecated`, it should be `hex.cmake` itself.
`version.cmake` itself has already converted to use `math()`, but you do have a point in `CMAKE_MODULE_PATH` not being set in all cases, and although `hex.cmake` itself could update `CMAKE_MODULE_PATH` when called, then we better avoid it, especially as this is used to determine correct Zephyr to use, based on versions.

> I'm going to propose a message(DEPRECATION ...) in cmake/hex.cmake, which is presented when it's included. I hope that suffices.

I believe that to be a good alternative in this case.